packages/server/api/controllers/movies.controller.js

```diff
@@ -245,6 +245,42 @@ const getFeaturedMovie = async (req, res) => {
   }
 };
 
+const getNewArrivals = async (req, res) => {
+  try {
+    const newArrivals = await knex('movies')
+      .join('categories', 'movies.category_id', 'categories.id')
+      .orderBy('movies.movie_year', 'desc')
+      .limit(5)
+      .select(
+        'movies.id',
+        'movies.category_id',
+        'categories.name as category_name',
+        'movies.title',
+        'movies.description',
+        'movies.image_location',
+        'movies.movie_year',
+      );
+
+    if (newArrivals.length === 0) {
+      throw new HttpError('No movies found in the database');
+    }
+
+    const latestMovies = newArrivals.map((latestMovie) => ({
+      movie_id: latestMovie.id,
+      category_id: latestMovie.category_id,
+      category_name: latestMovie.category_name,
+      title: latestMovie.title,
+      description: latestMovie.description,
+      image_location: latestMovie.image_location,
+      movie_year: latestMovie.movie_year,
+    }));
```

Comment:
this seems redundant to me, can you explain please? :)

Reply:
i did to modify newArrivals array into a modified format before sending the response but yeah now i realised that no need to that :D